_posts/2020-09-19-github-actions-with-faasd.md

```diff
@@ -0,0 +1,218 @@
+        run: |
+          docker run --rm -v $HOME:/root -w $HOME \
+          openfaas/faas-cli:latest-root \
+          login \
+          -p ${{ secrets.OPENFAAS_GATEWAY_PASSWD }} \
+          -g ${{ secrets.OPENFAAS_GATEWAY }} \
+          && \
+          sudo chown $USER:$USER -R $HOME/.openfaas && \
+          ls -lah $HOME
+      - 
+        name: Build and Push OpenFaaS function
+        run: |
+          # Update rights of build dir from root to current user \
+          # Build and push multi-arch image \
+          sudo chown $USER:$USER -R build/ && \
+          docker build \
+          --tag utsavanand2/faasd-hello:latest \
+          build/faasd-hello/ && \
+          docker push utsavanand2/faasd-hello:latest
+      - 
+        name: Deploying OpenFaaS function to your faasd appliance
+        run: |
+          docker run --rm -w `pwd` \
+          -v `pwd`:`pwd` \
+          -v $HOME:/root \
+          openfaas/faas-cli:latest-root \
+          deploy -f faasd-hello.yml
+```
+> Note: Replace the Docker UserID from the image tag name with your own.
+
+## Create and configure a Github Repo with Github Actions
+Initialize a git repo in the root directory of the project
+```sh
+# Initialize a git repo
+git init
+# Add a remote
+git remote add origin https://github.com/<Your-Github-Username>/faasd-hello.git
+# Stage and commit your changes
+git add .
+git commit -m "Initial Commit"
+```
+
+Create a Github Repo named faasd-hello and add the following secrets and their values to the repo for Github Actions to build push and deploy your OpenFaaS functions. Make sure that the secret names correspond with those defined in the github workflow YAML file as defined above.
+
+![secrets](/images/2020-09-19-github-actions-faasd/secrets.jpg)
+
+## Push the repo to Github to trigger Github Actions
+```sh
+git push -u origin master
+```
+
+## Check the status of your Github Actions build
+
+Under the `Actions` tab we can check the status of the workflow
+
+![workflow](/images/2020-09-19-github-actions-faasd/workflow.jpg)
+
+Since we have successfully built and deployed our functions let's invoke it
+
+## Invoke the function with curl
+
+```sh
+curl -d "Hello from faasd" http://$IP:8080/function/faasd-hello
+```
+
+![curl](/images/2020-09-19-github-actions-faasd/curl.jpg)
+
+## Closing thoughts
+
+OpenFaaS is best run on Kubernetes with faas-netes, but for smaller teams who want to jump on the serverless train and don't want to deal with the complexity of Kubernetes, faasd can be the obvious choice for them and with tools like Github Actions, repetitive tasks of testing, building, and deploying to OpenFaaS can be done with a simple `git push`
+
+# Take it further
+
+You can take this to the next level by running faasd on a Raspberry Pi and exposing it with [inlets](https://docs.inlets.dev) and leverage multi-arch builds with Docker's [Buildx](https://docs.docker.com/buildx/working-with-buildx/) with Github Actions.
```

Comment:
Explain how

Reply:
I was thinking of doing the same in the next blog post dedicated to multi-arch, and have a workflow that with a single build, deploys the same function at two places - one to my RPi (arm) k3s OpenFaas running faas-netes and another running as a faasd instance (amd64)